pkg/query/columnquery.go

```diff
@@ -740,16 +747,32 @@ func ComputeDiff(ctx context.Context, tracer trace.Tracer, base, compare profile
 			cols,
 			r.NumRows(),
 		))
+		compareTotals = append(compareTotals, compareTotal{
+			Cumulative:          math.Int64.Sum(cols[len(cols)-2].(*array.Int64)),
+			CumulativePerSecond: math.Float64.Sum(cols[len(cols)-1].(*array.Float64)),
+		})
 	}
 
-	for _, r := range base.Samples {
+	for i, r := range base.Samples {
 		func() {
 			columns := r.Columns()
+			totalCumulative := math.Int64.Sum(columns[len(columns)-4].(*array.Int64))
+			totalCumulativePerSecond := math.Float64.Sum(columns[len(columns)-3].(*array.Float64))
+
+			var cumulativeRatio, cumulativePerSecondRatio float64
+			if true { // configurable via UI
+				cumulativeRatio = float64(compareTotals[i].Cumulative) / float64(totalCumulative)
+				cumulativePerSecondRatio = compareTotals[i].CumulativePerSecond / totalCumulativePerSecond
+			} else {
+				cumulativeRatio = 1
+				cumulativePerSecondRatio = 1
+			}
+
 			cols := make([]arrow.Array, len(columns))
 			copy(cols, columns)
-			diff := multiplyInt64By(mem, columns[len(columns)-4].(*array.Int64), -1)
+			diff := multiplyInt64By(mem, columns[len(columns)-4].(*array.Int64), -1*cumulativeRatio)
 			defer diff.Release()
-			diffPerSecond := multiplyFloat64By(mem, columns[len(columns)-3].(*array.Float64), -1)
+			diffPerSecond := multiplyFloat64By(mem, columns[len(columns)-3].(*array.Float64), -1*cumulativePerSecondRatio)
```

Comment:
Right now, only this side of the comparison is changed. However, if this side has the higher total cumulative value, we need to "scale" it down. Meaning that we sometimes calculate new values with `1 * 0.95` which results in `0` for the int64 columns. 
In that case, we probably should instead "scale" the other side up, right? Basically making it a calculation of `1 * 1.05` which would still be `1` once casted back to `int64`.